File: Project/Project/microservices/notification-service/src/main/java/com/certifypro/notification/service/NotificationService.java
package com.certifypro.notification.service;

import com.certifypro.notification.dto.response.NotificationResponse;
import com.certifypro.notification.dto.response.PageResponse;
import com.certifypro.notification.entity.Notification;

/**
 * In-app notifications. Other modules trigger notifications via notifyUser / notifyRole.
 * The user-facing endpoints (list/read/create) are wired in the NotificationController.
 */
public interface NotificationService {

    Notification notifyUser(Long userId, String category, String message);

    /** Fan-out a notification to every user holding the given role. */
    void notifyRole(String role, String category, String message);

    PageResponse<NotificationResponse> listForUser(Long userId, int page, int limit);

    NotificationResponse markRead(Long id);
}

File: Project/Project/frontend/src/services/notificationService.js
import http from './http';

export async function list(params) {
  const res = await http.get('/notifications', { params });
  return res.data.data;
}
export async function markRead(id) {
  const res = await http.patch(`/notifications/${id}/read`);
  return res.data.data;
}
export async function create(body) {
  const res = await http.post('/notifications', body);
  return res.data.data;
}
